File: bindings/lua/Polycode/SceneLabel.lua
require "Polycode/ScenePrimitive"

class "SceneLabel" (ScenePrimitive)


function SceneLabel:__getvar(name)
	if name == "positionAtBaseline" then
		return Polycode.SceneLabel_get_positionAtBaseline(self.__ptr)
	end
	if ScenePrimitive["__getvar"] ~= nil then
		return ScenePrimitive.__getvar(self, name)
	end
end

function SceneLabel:__setvar(name,value)
	if name == "positionAtBaseline" then
		Polycode.SceneLabel_set_positionAtBaseline(self.__ptr, value)
		return true
	end
	if ScenePrimitive["__setvar"] ~= nil then
		return ScenePrimitive.__setvar(self, name, value)
	else
		return false
	end
end
function SceneLabel:SceneLabel(...)
	local arg = {...}
	if type(arg[1]) == "table" and count(arg) == 1 then
		if ""..arg[1].__classname == "ScenePrimitive" then
			self.__ptr = arg[1].__ptr
			return
		end
	end
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SceneLabel(unpack(arg))
	end
end

function SceneLabel:getText()
	local retVal =  Polycode.SceneLabel_getText(self.__ptr)
	return retVal
end

function SceneLabel:setLabelActualHeight(actualHeight)
	local retVal = Polycode.SceneLabel_setLabelActualHeight(self.__ptr, actualHeight)
end

function SceneLabel:getLabelActualHeight()
	local retVal =  Polycode.SceneLabel_getLabelActualHeight(self.__ptr)
	return retVal
end

function SceneLabel:getTextWidthForString(text)
	local retVal = Polycode.SceneLabel_getTextWidthForString(self.__ptr, text)
	return retVal
end

function SceneLabel:setText(newText)
	local retVal = Polycode.SceneLabel_setText(self.__ptr, newText)
end

function SceneLabel:updateFromLabel()
	local retVal =  Polycode.SceneLabel_updateFromLabel(self.__ptr)
end

function SceneLabel:__delete()
	if self then Polycode.delete_SceneLabel(self.__ptr) end
end

File: bindings/javascript/Polycode/SceneLabel.js
require('Polycode/ScenePrimitive')

function SceneLabel(text,size,fontName,amode,actualHeight) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SceneLabel(text,size,fontName,amode,actualHeight)
	}
	Object.defineProperties(this, {
		'positionAtBaseline': { enumerable: true, configurable: true, get: SceneLabel.prototype.__get_positionAtBaseline, set: SceneLabel.prototype.__set_positionAtBaseline}
	})
}


SceneLabel.prototype = Object.create(ScenePrimitive.prototype)

SceneLabel.prototype.__get_positionAtBaseline = function() {
	return Polycode.SceneLabel__get_positionAtBaseline(this.__ptr)
}

SceneLabel.prototype.__set_positionAtBaseline = function(val) {
	Polycode.SceneLabel__set_positionAtBaseline(this.__ptr, val)
}


SceneLabel.prototype.getText = function() {
	return Polycode.SceneLabel_getText(this.__ptr)
}

SceneLabel.prototype.setLabelActualHeight = function(actualHeight) {
	Polycode.SceneLabel_setLabelActualHeight(this.__ptr, actualHeight)
}

SceneLabel.prototype.getLabelActualHeight = function() {
	return Polycode.SceneLabel_getLabelActualHeight(this.__ptr)
}

SceneLabel.prototype.getTextWidthForString = function(text) {
	return Polycode.SceneLabel_getTextWidthForString(this.__ptr, text)
}

SceneLabel.prototype.setText = function(newText) {
	Polycode.SceneLabel_setText(this.__ptr, newText)
}

SceneLabel.prototype.updateFromLabel = function() {
	Polycode.SceneLabel_updateFromLabel(this.__ptr)
}
